app/src/main/java/org/mozilla/tv/firefox/settings/SettingsRepo.kt

```diff
@@ -0,0 +1,41 @@
+package org.mozilla.tv.firefox.settings
+
+import android.app.Application
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.content.SharedPreferences
+import android.os.StrictMode
+import android.preference.PreferenceManager
+import org.mozilla.tv.firefox.R
+
+private const val PREF_KEY_TELEMETRY = R.string.pref_key_telemetry
+
+class SettingsRepo(private val applicationContext: Application) {
+    private val _sharedPreferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
+    private val _resources = applicationContext.resources
+
+    private val _dataCollectionEnabled = MutableLiveData<Boolean>()
+    val dataCollectionEnabled: LiveData<Boolean> = _dataCollectionEnabled
+
+    init {
+        loadSettingsFromPreferences()
+    }
```

Comment:
This could be problematic if we want to test this repo in the future, since during unit tests we won't want to actually pull from disk.  Suggest moving this to a public setup function that gets called by the service locator.

Reply:
Hm, this is also writing to SharedPreferences, so I don't think we can avoid that here. Could we just mock the SharedPreferences from the context to fix this?

Reply:
Good point.  This is using `StrictMode` too, which is another Android class.  I think the cleanest way to get this testable would be to inject a 'persistence manager' interface, that just has a boolean getter and setter.  In prod it would be backed by `SharedPreferences`, in tests it could be an anonymous object.  Unless I'm mistaken that would let us get rid of references to `Application`, `StrictMode`, `Resources`, and `SharedPreferences`, meaning we could run the tests without Robolectric.

From a practical standpoint, that means the repo would be about ten lines long and wouldn't really even require testing.  If we don't expect Settings to grow in the future, it may be simpler to just leave it as is.